cl/citations/tests.py

```diff
@@ -355,39 +429,350 @@ def test_disambiguate_citations(self):
+                              'citation no-link"><span class="volume">22'
+                              '</span> <span class="reporter">U.S.</span> '
+                              '<span class="page">33</span> </span><pre class='
+                              '"inline">asdf</pre>')
+        test_pairs = [
+            # Simple example for full citations
+            ('asdf 22 U.S. 33 asdf', full_citation_html),
+
+            # Using a variant format for U.S. (Issue #409)
+            ('asdf 22 U. S. 33 asdf', full_citation_html),
+
+            # Full citation across line break
+            ('asdf John v. Doe, 123\nU.S. 456, upholding foo bar',
+             '<pre class="inline">asdf John v. Doe, </pre><span class="'
+             'citation no-link"><span class="volume">123</span>\n<span class='
+             '"reporter">U.S.</span> <span class="page">456</span></span><pre'
+             ' class="inline">, upholding foo bar</pre>'),
+
+            # First kind of short form citation (meaningless antecedent)
+            ('asdf. 515 U.S., at 240. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf. </span><span class="volume">515</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">240'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Second kind of short form citation (meaningful antecedent)
+            ('asdf, 1 U. S., at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Short form citation with no comma after reporter in original
+            ('asdf, 1 U. S. at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf, </span><span class="volume">1</span> '
+             '<span class="reporter">U.S.</span>, at <span class="page">2'
+             '</span></span><pre class="inline">. foobar</pre>'),
+
+            # Short form citation across line break
+            (u'asdf.’ ” 123 \n U.S., at 456. Foo bar foobar',
+             '<pre class="inline">asdf.’ </pre><span class="citation no-link">'
+             '<span class="antecedent">” </span><span class="volume">123'
+             '</span> \n <span class="reporter">U.S.</span>, at <span class='
+             '"page">456</span></span><pre class="inline">. Foo bar foobar'
+             '</pre>'),
+
+            # First kind of supra citation (standard kind)
+            ('asdf, supra, at 2. foobar',
+             '<pre class="inline"></pre><span class="citation no-link"><span '
+             'class="antecedent">asdf,</span><span> supra</span><span>, at '
```

Comment:
Seems like the span around `supra`, if it's worth keeping at all, should have  class? Larger point though: Why bother putting spans around the antecedents?

Reply:
This is just my bias for trying to enforce semantic markup everywhere. However, I agree that it's probably unnecessary here 😄

Reply:
Fixed (partially) in b1e76cd. It turns out that it's necessary to modify the general supra citation string at least a bit in order to prevent subsequent iterations of the regex/html replacement logic from trying to replace strings that have already been matched. So wrapping the antecedent guess in a `<span>` is one way to accomplish this; i.e., to mark that portion of the HTML as already replaced. Maybe there's an easier way to do this, but I don't think there's any harm in including that markup. (Unless we're trying to minimize bytes sent over the network... 😄)

Reply:
OK. It looks fixed to me.